internal/pkg/aws/session/session.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package session provides functions that return AWS sessions to use in the AWS SDK.
+package session
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+// Default returns a session configured against the "default" AWS profile.
+func Default() (*session.Session, error) {
+	return session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+}
+
+// FromProfile returns a session configured against the input profile name.
+func FromProfile(name string) (*session.Session, error) {
+	return session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+		Profile:           name,
+	})
+}
+
+// FromRole returns a session configured against the input role and region.
+func FromRole(roleARN string, region string) (*session.Session, error) {
```

Comment:
Could we detect the region from the role's ARN? Using [arn.Parse](https://github.com/aws/aws-sdk-go/blob/v1.25.7/aws/arn/arn.go#L61) for example

Reply:
Hmm I don't think so, but I'm not certain. As far as my understanding goes there is no region in the role ARNs.

Reply:
oh yeah, you're right!

Reply:
Love me some global resources :)